hivemind/utils/mpfuture.py

```diff
@@ -101,213 +134,133 @@ def _set_event_if_necessary(self):
         async def _event_setter():
             self._aio_event.set()
 
-        if self._loop.is_running() and loop == self.get_loop():
+        if self.get_loop().is_running() and loop == self.get_loop():
             asyncio.create_task(_event_setter())
-        elif self._loop.is_running() and loop != self.get_loop():
-            asyncio.run_coroutine_threadsafe(_event_setter(), self._loop)
+        elif self.get_loop().is_running() and loop != self.get_loop():
+            asyncio.run_coroutine_threadsafe(_event_setter(), self.get_loop())
         else:
-            self._loop.run_until_complete(_event_setter())
+            self.get_loop().run_until_complete(_event_setter())
 
     @classmethod
-    def _initialize_backend_if_necessary(cls):
+    def _initialize_mpfuture_backend(cls):
         pid = os.getpid()
-        if MPFuture._active_pid != pid:
-            with MPFuture._initialization_lock:
-                if MPFuture._active_pid != pid:
-                    # note: the second if is intentional, see https://en.wikipedia.org/wiki/Double-checked_locking
-                    logger.debug(f"Initializing MPFuture backend for pid {pid}")
-                    receiver_pipe, cls._process_wide_pipe = mp.Pipe(duplex=False)
-                    cls._active_pid, cls._active_futures, cls._status_requests = pid, {}, {}
-                    cls._pipe_waiter_thread = threading.Thread(
-                        target=cls._process_updates_in_background,
-                        args=[receiver_pipe],
-                        name=f"{__name__}.BACKEND",
-                        daemon=True,
-                    )
-                    cls._pipe_waiter_thread.start()
-
-    @classmethod
-    def reset_backend(cls):
-        """
-        Reset the MPFuture backend. This is useful when the state may have been corrupted
-        (e.g. killing child processes may leave the locks acquired and the background thread blocked).
+        logger.debug(f"Initializing MPFuture backend for pid {pid}")
+        assert pid != cls._active_pid, "already initialized"
```

Comment:
```suggestion
```

nit: I'm not sure this assert is necessary since this condition is already checked twice before calling this method :)